src/components/userProfile/userProfile.js

```diff
@@ -1,51 +1,84 @@
 import React from "react";
-import {getUserProfileInfo} from '../../services/userService';
+import {editUserInfo, getUserProfileInfo} from '../../services/userService';
 import Header from '../navigation/header';
-import {Button, Input, Layout} from 'antd';
+import {Button, Form, Input, Layout, Popconfirm} from 'antd';
 import {CheckOutlined, CloseOutlined, PlusCircleOutlined} from '@ant-design/icons';
 import NewCarModal from "../newCarModal/index";
+import {confirmMessage, errorMessage} from "../../services/alert.service";
+import {generalErrorMessages} from "../../constants/messages/general";
+import {inputValidationErrors} from "../../constants/messages/inputValidationErrors";
+import {userErrorMessages} from "../../constants/messages/user";
 
 /* now the class is using instead of function
  because it wasn't working with function,
  but it will be rewritten later
  */
 class UserProfilePage extends React.Component {
-
     constructor(props) {
         super(props);
         this.state = {
-            isModalOpen: this.props.isModalOpen,
-            userData: {
-                name: "",
-                surname: ""
-            }
+            isModalOpen: this.props.isModalOpen
         };
-    }
+    };
 
-    async componentDidMount() {
-        var userData = await getUserProfileInfo();
-        if (userData !== undefined) {
-            this.setState({
-                userData: userData
-            });
-        }
+    componentDidMount = async () => {
+        const userData = await getUserProfileInfo();
+        this.setState({userData: userData});
     };
 
     static getDerivedStateFromProps = (nextProps, prevState) => {
         return {
             isModalOpen: nextProps.isModalOpen
         };
-    }
+    };
+
+    onFinishFailed = () => {
+        errorMessage(
+            userErrorMessages.EDIT_USER_PROFILE_BLOCKED,
+            generalErrorMessages.CORRECT_ALL_COMMENTS
+        );
+    };
+
+    onFinish = (values) => {
+        confirmMessage().then((res) => {
+            if (res) {
+                editUserInfo(values);
```

Comment:
I think you should add a check if the fields aren't identical, because in this case it makes no sense to send a request